### ISSUE TITLE
Retrieving the username within CardDAV/Backend/PDO

### ISSUE THREAD

**Issue by wjgilmore** [fruux/sabre-dav#717]
Hi,

We're using sabre-dav with great success, however are running into a problem that I just can't seem to resolve. We'd like to retrieve the username within CardDAV/Backend/PDO.php (specifically our extended class), and I've been following along with this tutorial: http://sabre.io/dav/per-user-directories/.

I've extended the CardDAV/Backend/PDO.php constructor to accept the Sabre/DAV/Auth/Plugin object like so in our "server.php" file:

    $authBackend      = new NomoBackend($pdo);

    $authPlugin = new Sabre\DAV\Auth\Plugin($authBackend, 'SabreDAV');
    $principalBackend = new NomoPrincipalBackend($pdo);
    $carddavBackend   = new NomoCardDAVBackend($authPlugin, $pdo);
    ...
    $server->addPlugin($authPlugin);
    $server->exec();

I can absolutely confirm availability of the username inside the plugin (by way of a log message), however that `$authPlugin` object does not contain the username after being passed into the CardDAVBackend. Is there something I am misinterpreting regarding this tutorial? Does the Auth plugin have to be passed into one of the nodes which is subsequently passed into `Sabre\DAV\Server`?

Thanks a million,
Jason

**Comment by evert** [comment]
Hi Jason,

The reason you're not seeing it, is likely because you're attempting to access the value in the constructor. The username value gets populated a bit later.

By the time functions like getCalendarsForUser etc are called, the value will be there.

Only after `$server->exec()` gets called, the authentication systems kicks in.

**Comment by wjgilmore** [comment]
Hi Evert,

Oh wow oh wow that was indeed the problem. I was attempting to access the value too early. I can retrieve it no problem outside of the constructor. Is me AT evertpot.com your active e-mail address? I'd like to send you some $$ for a few drinks because you have saved me from likely numerous additional hours of insanity.

Jason

**Comment by evert** [comment]
Hey there!

Thanks so much =) There's no need to pay back though... It would be nice to have a little donation box for people to express their support in the future, but right now we're not really set up.

If you definitely feel inclined though, how about a 1-time donation to doctors without borders?

https://donate.doctorswithoutborders.org/onetime.cfm

Every bit helps =)

**Comment by wjgilmore** [comment]
Done this morning! Thanks again for the great project and assistance.

**Comment by evert** [comment]
Awesome! Really cool =)
